src/olympia/blocklist/migrations/0031_auto_20230612_1447.py

```diff
@@ -0,0 +1,28 @@
+# Generated by Django 4.2.1 on 2023-06-12 14:47
+
+from django.db import migrations
+
+
+def convert_min_max_to_block_version(apps, schema_editor):
+    Block = apps.get_model('blocklist', 'Block')
+    BlockVersion = apps.get_model('blocklist', 'BlockVersion')
+    File = apps.get_model('files', 'File')
+
+    for block in Block.objects.all():
+        block_versions = [
+            BlockVersion(version_id=version_id, block=block)
+            for version_str, version_id in File.objects.filter(
```

Comment:
As currently written, if the migration fails mid-way we're in trouble, because it can't pick up where it left off - the `bulk_create()` would fail on already created `BlockVersion`. Could you avoid them in this queryset ?

Reply:
if we assume that `bulk_create` doesn't fail in a way that some of the `BlockVersion`s are created and not others, then yeah we could filter out any `Block` that already has at least one `BlockVersion`.

Reply:
Or, as I now think you mean, filter them out in the `File` query.

Reply:
Yeah sorry, I meant filter them out of the `File` query.